### PR TITLE
Makes has_brain_worms cheaper

### PR DESCRIPTION
**Pull request by Fox-McCloud**
`has_brain_worms` gets called a whopping 1,000,000 times in a 2 hour shift...that's ridiculous.

Unfortunately, due to the nature of borers, this proc **has** to be utilized for a lot of common things...any time there's a medical hud update or whenever someone hears someone talking.

Either case, this makes the proc a lot cheaper by making it so that it doesn't loop through all of a mob's contents then do an istype check on each one until it finds the borer and returns the borer.

Now carbons just have a `borer` holder var that is checked for. No looping and no istype checking.

Additionally, for whatever reason, the medical hud checks for `has_brain_worms` twice. I haven't a clue as to why this is.

This also should help resolve hypothetical GC issues (haha, mobs....GC...ha) when a carbon with a borer is deleted.

### CODE REVIEW

File: code/game/gamemodes/changeling/powers/panacea.dm [hunk 1]
```diff
@@ -13,8 +13,6 @@
 
 	var/mob/living/simple_animal/borer/B = user.has_brain_worms()
 	if(B)
-		if(B.controlling)
-			B.detach()
```
Comment (Fox-McCloud) [inline code/game/gamemodes/changeling/powers/panacea.dm]: `leave_host` checks for controlling then calls `detach`, so this is redundant.

File: code/game/gamemodes/miniantags/borer/borer.dm [hunk 2]
```diff
@@ -564,7 +565,8 @@
 	host.reset_perspective(null)
 	host.machine = null
 
-	var/mob/living/H = host
+	var/mob/living/carbon/H = host
```
Comment (Fox-McCloud) [inline code/game/gamemodes/miniantags/borer/borer.dm]: Borer code is unspeakably bad...it's like the creator couldn't decide if you could only infest carbons, all living mobs, or only humans.

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -19,6 +19,10 @@
 /mob/living/carbon/Destroy()
 	QDEL_LIST(internal_organs)
 	QDEL_LIST(stomach_contents)
+	var/mob/living/simple_animal/borer/B = has_brain_worms()
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/carbon.dm]: we don't want to do:

```
if(borer)
	borer.leav_host()
	QDEL_NULL(borer)
```

Because then `borer` will be `null` and won't be `qdel`'d.

File: code/game/gamemodes/miniantags/borer/borer.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/game/gamemodes/miniantags/borer/borer.dm:568]: Greetings from the future, where I am scraping for borer fixes in a rework because when I wrote them I couldn't decide if you could only infest carbons, all living mobs, or only humans.
